presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -737,10 +744,11 @@ else if (!hasAnyFilter(columnIndex)) {
             }
         }
 
+        if (appendRowNumber) {
+            blocks[outputColumns.size()] = createRowNumbersBlock(positions, positionCount, this.getFilePosition());
```

Comment:
This is incorrect and will lead to issues.

Consider this case.
List<Page> pages;
pages.add(reader.nextPage());
pages.add(reader.nextPage());

Now the page1 and page2 shares the same underlying rowNumbersArray.

To prevent this, Presto implements BlockLeases. 

https://github.com/prestodb/presto/blob/master/presto-orc/src/main/java/com/facebook/presto/orc/reader/ListSelectiveStreamReader.java#L583

May be, in this PR, create long arrays always and do not reuse it, in the next PR, you can implement the block leases.

Reply:
Interesting, in the koski model, client is supposed to fetch the batch immediately as we re-use memory. I assumed the same here as buffers were re-used in the stream readers. Will look into block leases and how they work

Reply:
Okay, will integrate with block leases in next PR